Tests/DataPipeline/DataPipelineCompatibilityTests.swift

```diff
@@ -0,0 +1,438 @@
+@testable import CioDataPipelines
+@testable import CioInternalCommon
+import Foundation
+@testable import Segment
+import SharedTests
+import XCTest
+
+private typealias SavedEvent = [String: Any]
+
+class DataPipelineCompatibilityTests: UnitTest {
+    private var customerIO: CustomerIO!
+    private var analytics: Analytics!
+    private var storage: Storage!
+
+    private let deviceInfoMock = DeviceInfoMock()
+    private let eventBusHandlerMock = EventBusHandlerMock()
+    private let globalDataStoreMock = GlobalDataStoreMock()
+
+    override func setUp() {
+        super.setUp()
+
+        diGraphShared.override(value: dateUtilStub, forType: DateUtil.self)
+        diGraphShared.override(value: deviceInfoMock, forType: DeviceInfo.self)
+        diGraphShared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
+        diGraphShared.override(value: globalDataStoreMock, forType: GlobalDataStore.self)
+
+        customerIO = createCustomerIOInstance()
+
+        // setting up analytics for testing
+        analytics = customerIO.analytics
+        guard let analytics = analytics else {
+            fatalError("Analytics instance is nil. The SDK has been set up incorrectly.")
+        }
+        // wait for analytics queue to start emitting events
+        analytics.waitUntilStarted()
+        // get storage instance so we can read final events
+        storage = analytics.storage
+        storage.hardReset(doYouKnowHowToUseThis: true)
+    }
+
+    override func tearDown() {
+        customerIO.clearIdentify()
+        super.tearDown()
+    }
+
+    // MARK: profile
+
+    func test_identifyWithoutAttributes_expectFinalJSONHasCorrectKeysAndValues() {
+        let givenIdentifier = String.random
+
+        customerIO.identify(identifier: givenIdentifier)
+
+        let allEvents = readTypeFromStorage(key: Storage.Constants.events)
+        let filteredEvents = allEvents.filter { $0.eventType == "identify" }
+        XCTAssertEqual(filteredEvents.count, 1, "too many events received")
+
+        guard let savedEvent = filteredEvents.last else {
+            XCTFail("saved event must not be nil")
```

Comment:
bit confused here, the check before here verifies there is only `1` event, and this checks if it's not empty, wouldn't it be verified in the check before that?

Reply:
Unfortunately not exactly, because when the number is `0`, it may crash the tests. My [explanation in base PR](https://github.com/customerio/customerio-ios/pull/484#discussion_r1464606826) might help explain this better.